### PR TITLE
35-show all issues in db

### LINKED ISSUE #35: All the issues fit on a single page
## Summary

The user selects to view all issues.

## Acceptance criteria

```gherkin
Given the user is in the issues menu
And there are existing issues I-1 to I-10
When the user chooses to view all issues
Then the user will see issues I-1 to I-10 displayed on the screen
```

### PR DESCRIPTION
**Pull request by ebarylko**
## Summary

The user selects to view all issues once they are within the issues menu and sees the first 20 issues.

The issues will appear like this.

image

closes #35 

### src/main/kotlin/anttracker/issues/SetupSchema.kt
* Added the schema for `Contact`, `Issue`, `Release`, and `Product`

### build.gradle.kts
* Added dependencies for `Exposed`

### src/main/kotlin/anttracker/issues/Issues.kt
* Added `displayAllIssuesMenu` which shows all the issues from the database
* Added `mkIssuesMenu` which generates the issues menu

### src/main/kotlin/anttracker/issues/Screen.kt
* Changed `ScreenWithMenu` so it can take a customized prompt 

### src/main/kotlin/anttracker/Main.kt
* Added schema creation to the start of the program

### CODE REVIEW

File: src/main/kotlin/anttracker/Main.kt [hunk 1]
```diff
@@ -5,39 +5,43 @@ Rev 1 - 2024/07/01 Original by Micah
...
+import anttracker.contact.menu as contactMenu
+import anttracker.issues.mainIssuesMenu as issuesMenu
+import anttracker.product.menu as productMenu
+import anttracker.release.menu as releaseMenu
+import anttracker.request.menu as requestMenu
 
-class Terminal {
-    fun printLine() = println()
+fun main() {
+    Database.connect("jdbc:h2:mem:test;DB_CLOSE_DELAY=-1", driver = "org.h2.Driver")
```
Comment (micahdbak) [inline src/main/kotlin/anttracker/Main.kt:17]: Is this the line where we can configure the connection to SQLite or MySQL in the future?

File: src/main/kotlin/anttracker/issues/Issues.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+internal val issuesMenu = mkIssuesMenu()
+
+fun mainIssuesMenu() {
+    val t = Terminal()
+    var currentScreen: Screen? = issuesMenu
+
+    while (currentScreen != null) {
+        t.printLine("Press 0 to exit or * for the main menu")
```
Comment (micahdbak) [inline src/main/kotlin/anttracker/issues/Issues.kt]: Is "`0`" to exit the application entirely? Also should "`*`" be "`" to return to the main menu?

File: src/main/kotlin/anttracker/issues/SetupSchema.kt [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+import org.jetbrains.exposed.sql.javatime.datetime
+import org.jetbrains.exposed.sql.transactions.transaction
+import java.util.*
+
+fun setupSchema() {
+    transaction {
+        SchemaUtils.createMissingTablesAndColumns(Products, Issues, Releases)
+
+        (0..10).forEach { id ->
+            val productId = Products.insert { it[name] = "Product $id" } get Products.id
```
Comment (micahdbak) [inline src/main/kotlin/anttracker/issues/SetupSchema.kt:18]: Love the test data

File: src/main/kotlin/anttracker/issues/SetupSchema.kt [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+    var description by Issues.description
+    var product by Product referencedOn Issues.product
+    var anticipatedRelease by Release referencedOn Issues.anticipatedRelease
+    var creationDate by Issues.creationDate
+    var status by Issues.status
+    var priority by Issues.priority
+}
+
+data class Issue1(
```
Comment (micahdbak) [inline src/main/kotlin/anttracker/issues/SetupSchema.kt]: What's the purpose of `Issue1`?

**Review by micahdbak — COMMENTED** [review]
Changes look really good!! There are some merge conflicts that GitHub is listing, could you merge the main branch into this branch to resolve them? Once that's resolved, I feel good approving this PR

File: src/main/kotlin/anttracker/issues/SetupSchema.kt (hunk 4 above)
Comment (ebarylko) [reply src/main/kotlin/anttracker/issues/SetupSchema.kt]: Removed.

File: src/main/kotlin/anttracker/issues/Issues.kt (hunk 2 above)
Comment (ebarylko) [reply src/main/kotlin/anttracker/issues/Issues.kt]: Changed.

File: src/main/kotlin/anttracker/Main.kt (hunk 1 above)
Comment (ebarylko) [reply src/main/kotlin/anttracker/Main.kt:17]: Yes. This is where we will connect to `sqlite`.

**Comment by ebarylko** [comment]
Conflicts were resolved.

> Changes look really good!! There are some merge conflicts that GitHub is listing, could you merge the main branch into this branch to resolve them? Once that's resolved, I feel good approving this PR

**Review by micahdbak — APPROVED** [review]
Great! Feel free to merge, I'll bring your changes into my PR and keep at it

**Pull request merged**